app/src/ui/merge-branch/merge.tsx

```diff
@@ -85,22 +94,109 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
       return
     }
 
-    this.updateCommitCount(branch)
+    this.updateMergeStatus(branch)
   }
 
   private onFilterTextChanged = (filterText: string) => {
     this.setState({ filterText })
   }
 
   private onSelectionChanged = async (selectedBranch: Branch | null) => {
-    if (selectedBranch) {
+    if (selectedBranch != null) {
       this.setState({ selectedBranch })
-      await this.updateCommitCount(selectedBranch)
+      await this.updateMergeStatus(selectedBranch)
     } else {
-      this.setState({ selectedBranch, commitCount: 0 })
+      this.setState({ selectedBranch, commitCount: 0, mergeStatus: null })
     }
   }
 
+  private renderNewMergeInfo() {
+    const { currentBranch } = this.props
+    const { selectedBranch, mergeStatus, commitCount } = this.state
+
+    if (
+      mergeStatus == null ||
+      currentBranch == null ||
+      selectedBranch == null ||
+      currentBranch.name === selectedBranch.name ||
+      commitCount == null
+    ) {
+      return null
+    }
+
+    return (
+      <div className="merge-status-component">
+        <MergeStatusHeader status={this.state.mergeStatus} />
+        <p className="merge-info">
+          {this.renderMergeStatusMessage(
+            mergeStatus,
+            selectedBranch,
+            currentBranch,
+            commitCount
+          )}
+        </p>
+      </div>
+    )
+  }
+
+  private renderMergeStatusMessage(
+    mergeStatus: MergeResultStatus,
+    selectedBranch: Branch,
+    currentBranch: Branch,
+    commitCount: number
+  ): JSX.Element {
+    if (mergeStatus.kind === MergeResultKind.Loading) {
+      return (
+        <React.Fragment>
+          Checking for ability to merge automatically...
+        </React.Fragment>
+      )
+    }
+
+    if (mergeStatus.kind === MergeResultKind.Clean) {
+      const pluralized = commitCount === 1 ? 'commit' : 'commits'
+      return (
+        <React.Fragment>
+          This will merge
+          <strong>{` ${commitCount} ${pluralized}`}</strong>
+          {` `}
+          from
+          {` `}
+          <strong>{selectedBranch.name}</strong>
+          {` `}
+          into
+          {` `}
+          <strong>{currentBranch.name}</strong>
+        </React.Fragment>
+      )
+    }
+
+    if (mergeStatus.kind === MergeResultKind.Invalid) {
+      return (
+        <React.Fragment>
+          Unable to merge unrelated histories in this repository
+        </React.Fragment>
+      )
+    }
+
+    const count = mergeStatus.conflictedFiles
+    const pluralized = count === 1 ? 'file' : 'files'
+    return (
+      <React.Fragment>
+        There will be
+        <strong>{` ${count} conflicted ${pluralized}`}</strong>
+        {` `}
+        when merging
```

Comment:
same comment here with multi-line sentences and spaces

Reply:
Tidied up in 9a6459626edeb3b7e1c8cb94eac1107b55f03405